### PR TITLE
Rename FetchClients.fetchClients to ClientRepo.all, and refactor

### PR DESCRIPTION
**Pull request by botandrose**

### CODE REVIEW

File: haskell_scotty/ClientRepo.hs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            c = convert fc vc
+            d = convert fd vd
+            e = convert fe ve
+            f = convert ff vf
+            g = convert fg vg
+            h = convert fh vh
+            i = convert fi vi
+            j = convert fj vj
+            k = convert fk vk
+            l = convert fl vl
```
Comment (botandrose) [inline haskell_scotty/ClientRepo.hs]: @BartMassey Please tell me there's a better way than these 13 preceding lines!

Comment (BartMassey) [inline haskell_scotty/ClientRepo.hs]: Of course there is. Let me cons up a quick patch.

Comment (botandrose) [inline haskell_scotty/ClientRepo.hs]:
```haskell
convertFunction pair =
  convert (fst pair) (snd pair)
 
instance QueryResults Client where
  convertResults fields values =
    let v = map convertFunction (Data.List.zip fields values) in
      Client (!! v 0) (!! v 1) (!! v 2) (!! v 3) (!! v 4) (!! v 5) (!! v 6) (!! v 7) (!! v 8) (!! v 9) (!! v 10) (!! v 11)
```

This was as far as I got before I remembered that lists can't contain heterogeneous types, found out you cant map a list to a tuple, then subsequently bashed my face into the keyboard for ten minutes.

Comment (botandrose) [inline haskell_scotty/ClientRepo.hs]: Pretending that `v` was magically a tuple above, is there a way to apply a function to a tuple of arguments in Haskell? Something like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply ?

**Comment by botandrose** [comment]
This is probably as good as its gonna get for now, so I'm merging.

File: haskell_scotty/ClientRepo.hs (hunk 1 above)
Comment (BartMassey) [inline haskell_scotty/ClientRepo.hs]: There's a thing called "uncurry" that works on 2-tuples. There is no uncurry13, but you can define it in the obvious way :-)

uncurry13 f (a,b,c,d,e,f,g,h,i,j,k,l,m) = f a b c d e f g h i j k l m

There's some games we can play with `ExistentialTypes`, or like I say there's Template Haskell. I'll get to it at some point.

**Pull request merged**